File: backends/api-3ncount3r/view-models/encounter.view-model.ts
import { EncounterCreatureViewModel } from "./encounterCreature.view-model";

export class EncounterViewModel {
    id: string = '';
    name: string = '';
    creatures: EncounterCreatureViewModel[] = [];
    selectedParty: string = '';
    roundCount: number = 0;
    currentTurn: number = 0;
}

File: backends/DDD.3ncount3r.API/ViewModels/EncounterViewModel.cs
using System.Text.Json.Serialization;

namespace DDD._3ncount3r.API.ViewModels
{
  public class EncounterViewModel
  {
    [JsonPropertyName("id")]
    public string Id { get; set; }

    [JsonPropertyName("name")]
    public string Name { get; set; }

    [JsonPropertyName("creatures")]
    public IEnumerable<EncounterCreatureViewModel> Creatures { get; set; }

    [JsonPropertyName("selectedParty")]
    public string SelectedParty { get; set; }

    [JsonPropertyName("roundCount")]
    public int RoundCount { get; set; }

    [JsonPropertyName("currentTurn")]
    public int CurrentTurn { get; set; }
  }
}
